File: valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        
        int i = 0, j = s.size() - 1;
        
        while (i < j) {
            
            while (i < j && isalnum(s[i]) == 0) { ++i; }
            while (i < j && isalnum(s[j]) == 0) { --j; }
            
            if (toupper(s[i]) != toupper(s[j])) { return false; }
            
            ++i;
            --j;
        }
        
        return true;
    }
};

File: valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        
        char[] sArr = s.toCharArray();
        
        int i = 0, j = sArr.length - 1;
        
        while (i < j) {
            
            while (i < j && !Character.isLetterOrDigit(sArr[i])) { ++i; }
            while (i < j && !Character.isLetterOrDigit(sArr[j])) { --j; }
            
            if (Character.toLowerCase(sArr[i]) != Character.toLowerCase(sArr[j])) { return false; }
            
            ++i;
            --j;
        }
        
        return true;
    }
}